frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -305,6 +315,140 @@ pub mod pallet {
+                        print("No era staking points struct available even though we have information that contract was staked before. This is a bug!");
+                        EraStakingPoints {
+                            total: Zero::zero(),
+                            stakers: BTreeMap::<T::AccountId, BalanceOf<T>>::new(),
+                        }
+                    })
+            } else {
+                EraStakingPoints {
+                    total: Zero::zero(),
+                    stakers: BTreeMap::<T::AccountId, BalanceOf<T>>::new(),
+                }
+            };
+
+            // Ensure that we can add additional staker for the contract.
+            if !latest_era_staking_points.stakers.contains_key(&staker) {
+                ensure!(
+                    latest_era_staking_points.stakers.len()
+                        < T::MaxNumberOfStakersPerContract::get() as usize,
+                    Error::<T>::MaxNumberOfStakersExceeded
+                );
+            }
+
+            // Increment the staked amount.
+            latest_era_staking_points.total += bonded_value;
+            let entry = latest_era_staking_points
+                .stakers
+                .entry(staker.clone())
+                .or_insert(Zero::zero());
+            *entry += bonded_value;
+
+            ensure!(
+                *entry >= T::MinimumStakingAmount::get(),
+                Error::<T>::InsufficientStakingValue
+            );
+
+            // Update ledger and payee
+            Self::update_ledger(&staker, &ledger);
+
+            let current_era = Self::get_current_era();
+
+            // Update staked information for contract in current era
+            ContractEraStake::<T>::insert(
+                contract_id.clone(),
+                current_era,
+                latest_era_staking_points,
+            );
+
+            // Update total staked value in era
+            let mut era_reward = Self::get_era_total(current_era).unwrap_or(Default::default());
+            era_reward.staked += bonded_value;
+            PalletEraRewards::<T>::insert(current_era, era_reward);
+
+            // If contract wasn't claimed nor staked yet, insert current era as last claimed era.
+            // When calculating reward, this will provide correct information to the algorithm since nothing exists
+            // for this contract prior to the current era.
+            if !era_when_contract_last_staked.is_some() {
+                ContractLastClaimed::<T>::insert(contract_id.clone(), current_era);
+            }
+
+            // Check if we need to update era in which contract was last changed. Can avoid one write.
+            let contract_last_staked_change_needed =
+                if let Some(previous_era) = era_when_contract_last_staked {
+                    // if values aren't different, no reason to do another write
+                    previous_era != current_era
+                } else {
+                    true
```

Comment:
can you do insert here?
ContractLastStaked::<T>::insert(&contract_id, current_era);

Reply:
I do it a few lines below.

The reason I did it like this was to avoid an unnecessary write.
If the current era is the same as the previous era in which contract was staked, no need to mutate the storage.

Should I change the comment in the code to make it more clear?

Reply:
I understand intention. What I meant is remove returning true value and insert here on line 441

Reply:
I cannot do that since I check for an additional condition above:
previous_era != current_era

I don't want to have two inserts or nested if statements.